### ISSUE TITLE
Issue with git Organisation response

### ISSUE THREAD

**Issue by invincibleJai** [fabric8-launcher/launcher-planning#127]
For `/services/git/organizations` i observed different behaviour 
 ie 

Scenario 1:
for my account https://github.com/invincibleJai there are 3 orgs but in response we get is just one org `[fabric-ui]`

Scenario 2:
for @jyasveer's  account https://github.com/jyasveer there are 2 orgs but in response we get is no orgs `[ ]`

i don't remember authorizing  any org while registering with Openshift.io

Apart from that we also need git user name i.e jyasveer alongwith user's orgs as we show currently in OSIO wizard. It will enable Users without any org to create repos under that user

**Comment by quintesse** [comment]
@invincibleJai obtaining the list of organizations follows these GitHub API rules: https://developer.github.com/v3/orgs/#list-your-organizations

The important part here is: "This only lists organizations that your authorization allows you to operate on in some way (e.g., you can list teams with read:org scope, you can publicize your organization membership with user scope, etc.). "

So are you sure that  you actually have right s in those other organizations that you mention are missing from the list?

**Comment by invincibleJai** [comment]
@quintesse thanks for updating. For Org "fabric-ui", "rhdt-dev" i can see list teams with read:org scope, and have publicized my organization membership as Public. Same shows up here https://api.github.com/users/invinciblejai/orgs
 Similar was case for https://api.github.com/users/jyasveer/orgs , let me know if am missing something here

**Comment by quintesse** [comment]
@invincibleJai that's not the correct API call, `/users/invinciblejai/orgs` does indeed give you the list of _all_ the organizations you belong to but that will also include the ones that you don't have sufficient permissions for. It's the `/user/orgs` API call that's important.

**Comment by invincibleJai** [comment]
Ah i tried curl as below 

`curl -X GET \
  https://api.github.com/user/orgs \
  -H 'authorization: token git-token' \`

and response is as expected i.e all three orgs as below

`[
    {
        "login": "fabric8-ui",
        "id": 25244298,
        "url": "https://api.github.com/orgs/fabric8-ui",
        "repos_url": "https://api.github.com/orgs/fabric8-ui/repos",
        ...
    },
    {
        "login": "rhdt-dev",
        "id": 26006895,
        "url": "https://api.github.com/orgs/rhdt-dev",
        "repos_url": "https://api.github.com/orgs/rhdt-dev/repos",
        ...
    },
    {
        "login": "fabric8-analytics",
        "id": 28134954,
        "url": "https://api.github.com/orgs/fabric8-analytics",
        "repos_url": "https://api.github.com/orgs/fabric8-analytics/repos",
        ...
    }
]`

**Comment by quintesse** [comment]
Not sure this should be closed yet? For dinner train there still seems to be a difference in the expected vs the actual result

**Comment by gastaldi** [comment]
I thought that was already answered in the

>and response is as expected i.e all three orgs as below

part. Can you confirm @invincibleJai ?

**Comment by gastaldi** [comment]
>Apart from that we also need git user name i.e jyasveer alongwith user's orgs as we show currently in OSIO wizard. It will enable Users without any org to create repos under that user

I have already taked to @jyasveer that the /services/git/organizations will not return the user name as an organization because that doesn't make sense and should be handled in the UI

**Comment by invincibleJai** [comment]
> part. Can you confirm @invincibleJai ?

while consuming git services directly as below i get as expected 

`curl -X GET \
  https://api.github.com/user/orgs \
  -H 'authorization: token git-token' \`

and response is as expected i.e all three orgs as below

`[
    {
        "login": "fabric8-ui",
        "id": 25244298,
        "url": "https://api.github.com/orgs/fabric8-ui",
        "repos_url": "https://api.github.com/orgs/fabric8-ui/repos",
        ...
    },
    {
        "login": "rhdt-dev",
        "id": 26006895,
        "url": "https://api.github.com/orgs/rhdt-dev",
        "repos_url": "https://api.github.com/orgs/rhdt-dev/repos",
        ...
    },
    {
        "login": "fabric8-analytics",
        "id": 28134954,
        "url": "https://api.github.com/orgs/fabric8-analytics",
        "repos_url": "https://api.github.com/orgs/fabric8-analytics/repos",
        ...
    }
]`

but for forge-api i get empty list as orgs

```
curl -X GET \
  https://forge.api.prod-preview.openshift.io/api/services/git/organizations \
  -H 'authorization: Bearer prod-preview-token' \
  -H 'x-app: osio'
```

response is 

```
[ ]
```

**Comment by gastaldi** [comment]
@invincibleJai are you using the same token returned by auth.prod-preview.openshift.io in the test?

**Comment by invincibleJai** [comment]
> @invincibleJai are you using the same token returned by auth.prod-preview.openshift.io in the test?

yes i can pass to you in DM if it helps?

**Comment by invincibleJai** [comment]
i even tried with @jyasveer and @arunkumars08 there was diff in response

**Comment by invincibleJai** [comment]
@gastaldi one more observation if i pass 
```X-Git-Provider : GitHub``` as well in header as below
```
curl -X GET \
  https://forge.api.prod-preview.openshift.io/api/services/git/organizations \
  -H 'authorization: Bearer prod-preview-token' \
  -H 'x-app: osio'
  -H 'x-git-provider: GitHub'
```

response is 

`internal server error, 500`

and if i don't pass "X-Git-Provider  works" it works i.e 200 but returns empty list [], ideally "X-Git-Provider" is required token as per swagger so it shouldn't work without it.

**Comment by gastaldi** [comment]
@invincibleJai I see the error in the logs. 
```
org.jboss.weld.exceptions.UnsatisfiedResolutionException: WELD-001308: Unable to resolve any beans for Type: interface io.fabric8.launcher.service.git.api.GitServiceFactory; Qualifiers: [@javax.enterprise.inject.Default(), @io.fabric8.launcher.service.git.spi.GitProvider(value=GITHUB)]
```
We'll fix that and I'll let you know when it's ready

**Comment by invincibleJai** [comment]
Thanks @gastaldi :)

**Comment by gastaldi** [comment]
PR created: https://github.com/fabric8-launcher/launcher-backend/pull/155 will merge once the integration tests pass

**Comment by gastaldi** [comment]
@invincibleJai PR merged, can you confirm it works for you now?

**Comment by invincibleJai** [comment]
@gastaldi "X-Git-Provider" in header is working as expected. but response is still empty list for me for orgs

**Comment by arunkumars08** [comment]
> i even tried with @jyasveer and @arunkumars08 there was diff in response

Yes, @invincibleJai. I have tried with my prod-preview token and it gives me empty array list but the same thing with Github API url with Github token , I am able to see my list of organizations.

**Comment by animuk** [comment]
Hi @invincibleJai : As discussed, because you are trying with the prod-preview osio access token and there you do not have any org access , you are getting empty array. so, Please try to get required org access in the prod-preview and check again. We will get the right set of git organization
Closing this issue after confirming from you.
thanks,

![image](https://user-images.githubusercontent.com/11207106/37141858-68648ff6-22dc-11e8-8529-85b2129b4145.png)

**Comment by invincibleJai** [comment]
Thanks @animuk for help 👍
